Create v1

darwyncook opened the pull request:
This is version 1 which you should run through your unit tests.

jcf3 commented:
SS class returns an object as it does not have a print statement.

SS class does not check for valid SS; must only contain numbers, none can be all zero, must be entered withv dashes and the correct number of numbers(ex: xxx-xx-xxxx), the first strand of SS can not be 666 or in range(900, 1000), and finally it can not be 78-05-1120 as that was used for advertising.

Email class does not validate email; must look for valid gTLD (ex: com, gov, edu, org, mil or net), must only contain one "@" and no other special characters.

Hash class should create a hash not solely dependent on password length, the hash equal operator shouldn't add a plus one and should check to make sure that is true

darwyncook commented:
I committed a version 2 of my code

jcf3 commented:
An invalid email does not raise InvalidEmail as it is looking for a ValueError to raise InvalidEmail.

darwyncook commented:
How about this?

import hashlib
import re


class InvalidEmail(Exception):
    pass


class InvalidSocial(Exception):
    pass


class InvalidPassword(Exception):
    pass

"""
class Email:

    def __init__(self, email):
        try:
            a, b = email.split("@")
            z = b.split(".")
            if z[-1] in ["com", "net", "edu"]:
                self.email = email
            else:
                raise InvalidEmail
        except ImportError:
            raise InvalidEmail

    def __str__(self):
        return self.email
"""

class Email:

    def __init__(self, email):
        search_Obj = re.search(r'[a-zA-Z]([a-zA-Z0-9]*)@([a-zA-Z0-9]*)\.edu|biz|com|gov', email, flags=0)
        if search_Obj is not None:
            self.email = search_Obj.group()
        else:
            raise InvalidEmail

    def __str__(self):
        return self.email


class SS:

    def __init__(self, social):
        search_Obj = re.search(r'[0-9]{3}-[0-9]{2}-[0-9]{4}', social, flags=0)
        if search_Obj is not None:
            self.social = search_Obj.group()
        else:
            raise InvalidSocial

    def __str__(self):
        return self.social


class Hash:
    def __init__(self, passwd):
        if len(passwd) > 7:
            self.hash = self.generate_hash(passwd)
        else:
            raise InvalidPassword

    def __eq__(self, other):
        if isinstance(other, str):
            return self.hash == self.generate_hash(other)
        elif isinstance(other, Hash):
            return self.hash == other.hash
        else:
            return NotImplemented

    @staticmethod
    def generate_hash(passwd):
        """ This is NOT secure, a random salt should be added"""
        return hashlib.sha512(passwd.encode("utf-8")).hexdigest()

jcf3 commented:
Email still does not return the correct string

darwyncook commented:
import hashlib
import re


class InvalidEmail(Exception):
    pass


class InvalidSocial(Exception):
    pass


class InvalidPassword(Exception):
    pass


class Email:

    def __init__(self, email):
        searchobj = re.search(r'^[a-zA-Z]([a-zA-Z0-9]*)@[a-zA-Z]([a-zA-Z0-9\.]*)\.(edu|biz|com|gov)$', email, flags=0)
        if searchobj is not None:
            self.email = searchobj.group()
        else:
            raise InvalidEmail

    def __str__(self):
        return self.email


class SS:

    def __init__(self, social):
        search_obj = re.search(r'^(?!219-09-9999|078-05-1120)(?!666|000|9\d{2})\d{3}-(?!00)\d{2}-(?!0{4})\d{4}$',
                               social, flags=0)
        if search_obj is not None:
            self.social = search_obj.group()
        else:
            raise InvalidSocial

    def __str__(self):
        return self.social


class Hash:
    def __init__(self, passwd):
        if len(passwd) > 7:
            self.hash = self.generate_hash(passwd)
        else:
            raise InvalidPassword

    def __eq__(self, other):
        if isinstance(other, str):
            return self.hash == self.generate_hash(other)
        elif isinstance(other, Hash):
            return self.hash == other.hash
        else:
            return NotImplemented

    @staticmethod
    def generate_hash(passwd):
        """ This is NOT secure, a random salt should be added"""
        return hashlib.sha512(passwd.encode("utf-8")).hexdigest()
